packages/runtimes/js/src/JSRuntime.js

```diff
@@ -475,20 +475,33 @@ function getLoaderRuntime({
     loaderCode = `(${loaderCode})`;
   }
 
+  let needsTierPrelude = false;
   if (mainBundle.type === 'js') {
     let parcelRequire = bundle.env.shouldScopeHoist
       ? 'parcelRequire'
       : 'module.bundle.root';
     loaderCode += `.then(() => ${parcelRequire}('${bundleGraph.getAssetPublicId(
       bundleGraph.getAssetById(bundleGroup.entryAssetId),
     )}'))`;
+
+    if (
+      options.featureFlags.tieredImports &&
+      (dependency.meta.kind === 'DeferredForDisplayTierImport' ||
+        dependency.meta.kind === 'DeferredTierImport')
+    ) {
+      loaderCode = `tier(${loaderCode})`;
+      needsTierPrelude = true;
+    }
   }
 
   let code = [];
 
   if (needsEsmLoadPrelude) {
     code.push(`let load = require('./helpers/browser/esm-js-loader');`);
   }
+  if (needsTierPrelude) {
+    code.push(`let tier = require('./helpers/browser/tier-loader');`);
```

Comment:
This code only exists for now in "lazy" priority code (same as doing a dynamic `import('x')`), this is one of the things that will need to change when we start thinking about "tiered" code possibly existing inline or external.